eth_worker/eth_manager/blockchain_sync/blockchain_sync.py

```diff
@@ -0,0 +1,145 @@
+            floor = max_enqueued_block + (chunk * sync_const.BLOCKS_PER_REQUEST)
+            ceiling = max_enqueued_block + ((chunk + 1) * sync_const.BLOCKS_PER_REQUEST)
+            if ceiling > latest_block:
+                ceiling = latest_block
+            # filter_job objects are just filter objects with floors/ceilings set
+            job = {
+                'filter_id': f.id,
+                'floor': floor,
+                'ceiling': ceiling
+            }
+            red.rpush(sync_const.THIRD_PARTY_SYNC_JOBS, json.dumps(job))
+        if ceiling:
+            # Sometimes there won't be a ceiling, if there's no new chunks since the last time it ran
+            persistence_module.set_filter_max_block(f.id, ceiling)
+        # With the jobs set, we can now start processing them
+        process_all_chunks()
+
+
+# Iterates through all jobs made by synchronize_third_party_transactions
+# Gets and processes them all!
+def process_all_chunks():
+    for filter_job in red.lrange(sync_const.THIRD_PARTY_SYNC_JOBS, 0, -1):
+        task = json.loads(filter_job)
+        filter = persistence_module.get_filter(task['filter_id'])
+        transaction_history = get_blockchain_transaction_history(
+            filter.contract_address, 
+            task['floor'], 
+            task['ceiling'], 
+            filter.filter_parameters,
+            filter.id
+        )
+        for transaction in transaction_history:
+            handle_transaction(transaction)
+        # Once a batch of chunks is completed, we can mark them completed
+        persistence_module.set_block_range_status(task['floor'], task['ceiling'], 'SUCCESS')
+        # Only pop the list (delete job from queue) after success
+        red.lpop(sync_const.THIRD_PARTY_SYNC_JOBS)
+
+# Processes newly found transaction event
+# Creates database object for transaction
+# Calls webhook
+# Sets sync status (whether or not webhook was successful)
+# Fallback if something goes wrong at this level: `is_synchronized_with_app` flag. Can batch unsynced stuff
+def handle_transaction(transaction):
+    # Check if transaction already exists (I.e. already synchronized, or first party transactions)
+    transaction_object = persistence_module.get_transaction(hash=transaction.transactionHash.hex())
+    if transaction_object and transaction_object.is_synchronized_with_app:
+        return True
+    transaction_object = persistence_module.create_external_transaction(
+        status = 'SUCCESS',
+        block = transaction.blockNumber,
+        hash = str(transaction.transactionHash.hex()),
+        contract_address = transaction.address,
+        is_synchronized_with_app = False,
+        recipient_address = transaction.args['to'],
+        sender_address = transaction.args['from'],
+        amount = transaction.args['value']
+    )
+
+    call_webhook(transaction_object)
+    # Transactions which we fetched, but couldn't sync for whatever reason won't be marked as completed
```

Comment:
It was an implicit branch, since `call_webhook` could error out and then `mark_transaction_as_completed` wouldn't be reached. Clarified it in the code a bit.